values/src/main/proto/spine/net/url.proto

```diff
@@ -0,0 +1,76 @@
+syntax = "proto3";
+
+package spine.net;
+
+option java_generate_equals_and_hash = true;
+option java_multiple_files = true;
+option java_package = "org.spine3.net";
+option java_outer_classname = "UrlProto";
+
+message Url {
+    oneof value {
+        string raw = 1;
+        Record record = 2;
+    }
+
+    // Structurized record
+    message Record {
+
+        // Application layer protocol
+        Protocol protocol = 1;
+
+        // User's email and password (both optional)
+        Authorization auth = 2;
+
+        // Domain name
+        string host = 3;
+
+        // Server's port (e.g. :8080)
+        // We leave port as String, because port 0 is hypothetically possible to use
```

Comment:
Because we'll have default value 0 for port then. Usually, if port is not set, it's 8080. If we leave 0 as "value not set" case, we won't be able to set port 0 explicitly